src/riak_core_bucket.erl

```diff
@@ -42,6 +42,11 @@
 -include_lib("eunit/include/eunit.hrl").
 -endif.
 
+-type riak_core_ring() :: riak_core_ring:riak_core_ring().
+-type bucket_type()  :: binary().
```

Comment:
Instead of `binary()`, how about `riak_core_bucket_type:bucket_type()` (like what you did for `riak_core_ring()`).

(Would be nice if we could also do the same thing for `bucket_props()` but it looks like that type is currently defined in a riak_kv module, so I guess we'll have to define it separately since it's not a dependency.)